server/app/repository/ApplicationRepository.java

```diff
@@ -76,27 +77,37 @@ private Application submitApplicationInternal(
               .eq("applicant.id", applicant.id)
               .eq("program.name", program.getProgramDefinition().adminName())
               .findList();
-      Optional<Application> completedApplication = Optional.empty();
-      for (Application application : oldApplications) {
-        // Delete any in-progress drafts, and mark obsolete any old applications.
-        if (application.getLifecycleStage().equals(LifecycleStage.DRAFT)) {
-          application.setLifecycleStage(LifecycleStage.ACTIVE);
-          application.setSubmitTimeToNow();
-          completedApplication = Optional.of(application);
-        } else {
-          application.setLifecycleStage(LifecycleStage.OBSOLETE);
-        }
-        application.save();
+      List<Application> latestDraft =
+          oldApplications.stream()
+              .filter(app -> app.getLifecycleStage().equals(LifecycleStage.DRAFT))
+              .collect(Collectors.toList());
```

Comment:
suggestion (blocking): prefer immutable collections from guava as a general rule
```suggestion
      ImmutableList<Application> latestDraft =
          oldApplications.stream()
              .filter(app -> app.getLifecycleStage().equals(LifecycleStage.DRAFT))
              .collect(ImmutableList.toImmutableList());
```